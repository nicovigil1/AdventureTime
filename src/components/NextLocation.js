import React from 'react'

export default class NextLocation extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      currentLocation: null 
    }
  }
  
  componentDidMount() {
    this.getCurrentQuest()
  }

  parseLocation = (location) => {
    return location.split("|")[1].split(",")[0]
  }
  getCurrentQuest = () => {
    const currentQuest = this.props.quest
    console.log(this.props.quest.current_location)
    if (currentQuest.current_location === 1) {
      this.setState({currentLocation: this.parseLocation(currentQuest.location_1)})
    } else if (currentQuest.current_location === 2) {
      this.setState({currentLocation: this.parseLocation(currentQuest.location_2)})
    } else if (currentQuest.current_location === 3) {
      this.setState({currentLocation: this.parseLocation(currentQuest.location_3)})
    }
    console.log(this.state.currentLocation)
  }

  goBack = () => {
    this.props.goBack({loginDisplay: true})
  }

  render () {
    return (
      <div style={center}>
        <button onClick={this.goBack}>Go back</button>
        <h1>Next location:</h1>
          <a href={"https://maps.google.com/?q=" + this.state.currentLocation}>{this.state.currentLocation}</a>
      </div>
    )
  }
}

const center = {
  display: "flex",
  flexDirection: "column"
}