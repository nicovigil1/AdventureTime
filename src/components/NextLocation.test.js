import React from 'react'
import NextLocation from './NextLocation'
import { shallow } from 'enzyme'
import { isMainThread } from 'worker_threads';


describe('Header', () => {
  let wrapper
  let quest = {}
  let goBack = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<NextLocation quest={quest} getUser={goBack} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})