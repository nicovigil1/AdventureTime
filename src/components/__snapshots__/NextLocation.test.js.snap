// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button>
    Check In
  </button>
  <h1>
    Next location:
  </h1>
  <a
    href="https://maps.google.com/?q=null"
  />
</div>
`;
